wsmaster/che-core-sql-schema/src/main/resources/che-schema/5.4.0/1__add_factory.sql

```diff
@@ -0,0 +1,164 @@
+--
+-- Copyright (c) 2012-2017 Codenvy, S.A.
+-- All rights reserved. This program and the accompanying materials
+-- are made available under the terms of the Eclipse Public License v1.0
+-- which accompanies this distribution, and is available at
+-- http://www.eclipse.org/legal/epl-v10.html
+--
+-- Contributors:
+--   Codenvy, S.A. - initial API and implementation
+--
+
+-- Factory button --------------------------------------------------------------
+CREATE TABLE che_factory_button (
+    id          BIGINT          NOT NULL,
+    type        VARCHAR(255),
+    color       VARCHAR(255),
+    counter     BOOLEAN,
+    logo        VARCHAR(255),
+    style       VARCHAR(255),
+
+    PRIMARY KEY (id)
+);
+--------------------------------------------------------------------------------
+
+
+-- Factory action --------------------------------------------------------------
+CREATE TABLE che_factory_action (
+    entity_id       BIGINT          NOT NULL,
+    id              VARCHAR(255),
```

Comment:
not sure it has to be not null, 
https://github.com/eclipse/che/blob/0e8b21ad3ff5da80a7a03ddf3f0b281cc3594dcb/core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/factory/Action.java#L23-L23